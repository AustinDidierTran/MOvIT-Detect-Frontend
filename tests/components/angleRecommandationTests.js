import React from 'react';
import TestUtils from 'react-dom/test-utils';
import PropTypes from 'prop-types';
import AngleRecommandation from '../../src/components/angleRecommendation';

describe('AngleRecommandation Tests', () => {
  it('should have proptypes', function () {
    // Actual value
    const actualValue = AngleRecommandation.WrappedComponent.propTypes;

    // Expected value
    const expectedValue = {
      language: PropTypes.string.isRequired,
      recActive: PropTypes.bool.isRequired,
      title: PropTypes.string.isRequired,
      maxAngle: PropTypes.number.isRequired,
      value: PropTypes.number,
      onChangeActive: PropTypes.func.isRequired,
      onChangeValue: PropTypes.func.isRequired,
    };


    // Test
    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });
});
