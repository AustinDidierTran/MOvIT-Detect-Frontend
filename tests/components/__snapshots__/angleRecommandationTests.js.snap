// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AngleRecommandation Tests should match the snapshot when the recommandation is active 1`] = `
<div>
  <div
    className="col-sm-12"
  >
    <div
      className="col-sm-4"
      style={
        Object {
          "paddingLeft": "0",
        }
      }
    >
      <Checkbox
        checked={true}
        className={null}
        disabled={false}
        id={null}
        inputId="activeRecCheck"
        label="Test"
        name={null}
        onChange={[Function]}
        onContextMenu={null}
        onMouseDown={null}
        readOnly={false}
        style={null}
        value={null}
      />
      <label
        htmlFor="activeRecCheck"
      >
        Test
      </label>
    </div>
  </div>
  <SliderValue
    max={45}
    min={0}
    onChange={[Function]}
    title="recommendations.angle.en"
    unit="Â°"
    value={0}
  />
</div>
`;

exports[`AngleRecommandation Tests should match the snapshot when the recommandation is inactive 1`] = `
<div>
  <div
    className="col-sm-12"
  >
    <div
      className="col-sm-4"
      style={
        Object {
          "paddingLeft": "0",
        }
      }
    >
      <Checkbox
        checked={false}
        className={null}
        disabled={false}
        id={null}
        inputId="activeRecCheck"
        label="Test"
        name={null}
        onChange={[Function]}
        onContextMenu={null}
        onMouseDown={null}
        readOnly={false}
        style={null}
        value={null}
      />
      <label
        htmlFor="activeRecCheck"
      >
        Test
      </label>
    </div>
  </div>
</div>
`;
